modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ScalaCaskServerCodegen.java

```diff
@@ -0,0 +1,958 @@
+package org.openapitools.codegen.languages;
+
+import io.swagger.v3.oas.models.OpenAPI;
+import io.swagger.v3.oas.models.media.Schema;
+import io.swagger.v3.oas.models.tags.Tag;
+import org.apache.commons.io.FileUtils;
+import org.openapitools.codegen.*;
+import org.openapitools.codegen.model.ModelMap;
+import org.openapitools.codegen.model.ModelsMap;
+import org.openapitools.codegen.model.OperationsMap;
+import org.openapitools.codegen.serializer.SerializerUtils;
+import org.openapitools.codegen.utils.ModelUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+import java.util.regex.PatternSyntaxException;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.openapitools.codegen.utils.StringUtils.camelize;
+
+public class ScalaCaskServerCodegen extends AbstractScalaCodegen implements CodegenConfig {
+    public static final String PROJECT_NAME = "projectName";
+
+    private final Logger LOGGER = LoggerFactory.getLogger(ScalaCaskServerCodegen.class);
+
+    public CodegenType getTag() {
+        return CodegenType.SERVER;
+    }
+
+    public String getName() {
+        return "scala-cask";
+    }
+
+    public String getHelp() {
+        return "Generates a scala-cask server.";
+    }
+
+
+    static String env(String key, String defaultValue) {
+        final String value = System.getenv().get(key);
+        return value == null || value.isEmpty() ? defaultValue : value;
+    }
+
+    static boolean debug = Boolean.parseBoolean(env("DEBUG", "false"));
+
+    protected String groupId = env("GROUP_ID", "org.openapitools");
+    protected String artifactId = env("ARTIFACT_ID", "caskgen");
+    protected String basePackage = groupId;
+    protected String artifactVersion = "1.0.0";
+    protected String gitUserId = env("GIT_USER", System.getProperty("user.name"));
+    protected String gitRepoId = env("GIT_REPO", artifactId);
+    protected String appName = env("APP_NAME", "Cask App");
+    protected String infoEmail = env("INFO_EMAIL", "contact@kindservices.co.uk");
```

Comment:
for gitUserId and gitRepoId,  you can use these directly from default codegen.

please do not use environment variables, and use additional properties instead. e.g.  https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ScalaAkkaHttpServerCodegen.java#L174-L224


infoEmail should be obtained from the spec so I don't think we need to set it here

Reply:
Many thanks.

I've reworked these properties and cleaned up a lot of the debug code.

I've changed the sample properties and output destination -- I hope that's not too confusing to follow from the original PR